cmd/broadcast/utils.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"bufio"
+	"encoding/hex"
+	ida "github.com/harmony-one/libunison/internal/ida/coopcast"
+	"io"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const pubKeySize int = 20
+
+// Entry is a single config of a node.
+type Entry struct {
+	Sid     string
+	IP      string
+	TCPPort string
+	UDPPort string
+	PubKey  string
+	Role    string
+}
+
+// Config is a struct containing multiple Entry of all nodes.
+type Config struct {
+	config []Entry
+}
+
+// NewConfig returns a pointer to a Config.
+func NewConfig() *Config {
+	config := Config{}
+	return &config
+}
+
+// GetPeerInfo returns the selfPeer, peerList, allPeers from config instance, which used to create node instance
+func (config *Config) GetPeerInfo() (ida.Peer, []ida.Peer, []ida.Peer) {
```

Comment:
Please consider using named return, to distinguish the three confusingly similar return values:

    func (config *Config) GetPeerInfo() (selfPeer ida.Peer, peerList []ida.Peer, allPeers []ida.Peer) {

As an added bonus, it will also simplify the code, because you won't have to declare `selfPeer`, `peerList`, and `allPeers` in separate `var` lines.

For more info, see https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters.

Reply:
done